apps/artwork_purchase/client/purchase_signup_form.coffee

```diff
@@ -1,37 +1,46 @@
 _ = require 'underscore'
+Q = require 'bluebird-q'
 Backbone = require 'backbone'
 Form = require '../../../components/mixins/form.coffee'
 mediator = require '../../../lib/mediator.coffee'
 analyticsHooks = require '../../../lib/analytics_hooks.coffee'
-User = require '../../../models/user.coffee'
+CurrentUser = require '../../../models/current_user.coffee'
 sanitizeRedirect = require 'artsy-passport/sanitize-redirect'
 Mailcheck = require '../../../components/mailcheck/index.coffee'
 Cookies = require 'cookies-js'
+LoggedOutUser = require '../../../models/logged_out_user.coffee'
 
 module.exports = class PurchaseSignupForm extends Backbone.View
   _.extend @prototype, Form
 
-  initialize: ({ @user }) -> #
+  initialize: -> #
+    @loggedOutUser = new LoggedOutUser
 
   submit: ({ success, error, isWithAccountCallback }) ->
-    @user.set (data = @serializeForm())
-    @user.prepareForInquiry().then =>
-      if @user.isWithAccount()
+    @loggedOutUser.set (data = @serializeForm())
+    @loggedOutUser.prepareForInquiry().then =>
+
+    userLookup = @loggedOutUser.findOrCreate silent: true
+      .then =>
+        Q.promise (resolve) =>
```

Comment:
this one actually needs it because you want it to resolve on error which it doesn't normally do (the case when the account lookup fails because the user does not exist). The same logic is used in the `prepareForInquiry` method.